vm/rust/src/lib.rs

```diff
@@ -380,65 +443,99 @@ fn build_block_context(
+            eth_l1_gas_price: NonZeroU128::new(felt_to_u128(gas_price_wei_felt))
+                .unwrap_or(default_gas_price),
+            strk_l1_gas_price: NonZeroU128::new(felt_to_u128(gas_price_fri_felt))
+                .unwrap_or(default_gas_price),
+            eth_l1_data_gas_price: NonZeroU128::new(felt_to_u128(data_gas_price_wei_felt))
+                .unwrap_or(default_gas_price),
+            strk_l1_data_gas_price: NonZeroU128::new(felt_to_u128(data_gas_price_fri_felt))
+                .unwrap_or(default_gas_price),
         },
         use_kzg_da: block_info.use_blob_data == 1,
-    }, ChainInfo{
-        chain_id: ChainId(chain_id_str.to_string()),
+    };
+    let chain_info = ChainInfo {
+        chain_id: ChainId::from(chain_id_str.to_string()),
         fee_token_addresses: FeeTokenAddresses {
             // both addresses are the same for all networks
-            eth_fee_token_address: ContractAddress::try_from(StarkHash::try_from("0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7").unwrap()).unwrap(),
-            strk_fee_token_address: ContractAddress::try_from(StarkHash::try_from("0x04718f5a0fc34cc1af16a1cdee98ffb20c31f5cd61d6ab07201858f4287c938d").unwrap()).unwrap(),
+            eth_fee_token_address: ContractAddress::try_from(
+                StarkHash::from_hex(
+                    "0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7",
+                )
+                .unwrap(),
+            )
+            .unwrap(),
+            strk_fee_token_address: ContractAddress::try_from(
+                StarkHash::from_hex(
+                    "0x04718f5a0fc34cc1af16a1cdee98ffb20c31f5cd61d6ab07201858f4287c938d",
+                )
+                .unwrap(),
+            )
+            .unwrap(),
         },
-    }, constants).unwrap()
-}
-
-
+    };
 
+    pre_process_block(state, old_block_number_and_hash, block_info.block_number).unwrap();
+    BlockContext::new(block_info, chain_info, constants, BouncerConfig::max())
+}
 
 lazy_static! {
     static ref CONSTANTS: HashMap<String, VersionedConstants> = {
         let mut m = HashMap::new();
-        m.insert("0.13.0".to_string(), serde_json::from_slice(include_bytes!("../versioned_constants_13_0.json")).unwrap());
-        m.insert("0.13.1".to_string(), serde_json::from_slice(include_bytes!("../versioned_constants_13_1.json")).unwrap());
+        m.insert(
+            "0.13.0".to_string(),
+            serde_json::from_slice(include_bytes!("../versioned_constants_13_0.json")).unwrap(),
+        );
+        m.insert(
+            "0.13.1".to_string(),
+            serde_json::from_slice(include_bytes!("../versioned_constants_13_1.json")).unwrap(),
+        );
+        m.insert(
+            "0.13.1.1".to_string(),
+            serde_json::from_slice(include_bytes!("../versioned_constants_13_1_1.json")).unwrap(),
+        );
```

Comment:
So the default version constants are for 0.13.2? What about blocks which are older than 0.13.0?

Reply:
> So the default version constants are for 0.13.2?

yes

> What about blocks which are older than 0.13.0?

Here is how we choose correct versioned constants:
```
if let Some(constants) = unsafe { &CUSTOM_VERSIONED_CONSTANTS } {
        constants.clone()
    } else if version < StarknetVersion::from_str(&"0.13.1").unwrap() {
        CONSTANTS.get(&"0.13.0".to_string()).unwrap().to_owned()
    } else if version < StarknetVersion::from_str(&"0.13.1.1").unwrap() {
        CONSTANTS.get(&"0.13.1".to_string()).unwrap().to_owned()
    } else if version < StarknetVersion::from_str(&"0.13.2").unwrap() {
        CONSTANTS.get(&"0.13.1.1".to_string()).unwrap().to_owned()
    } else {
        VersionedConstants::latest_constants().to_owned()
    }
```